cadence/contracts/PrizeSavings.cdc

```diff
@@ -1005,16 +1820,34 @@ access(all) contract PrizeSavings {
             panic("No request to pop")
         }
         
+        /// Returns the captured weight snapshot.
```

Comment:
This could potentially borrow a reference instead, so it doesn't need to copy the entire struct. In `selectWinners`, you iterate through it anyway, but you can just do that from the reference instead of a copy